fv3core/stencils/d_sw.py

```diff
@@ -93,8 +94,92 @@ def horizontal_relative_vorticity_from_winds(
         vorticity[0, 0, 0] = rarea * (vt - vt[0, 1, 0] - ut + ut[1, 0, 0])
 
 
+@gtscript.function
+def add_dw(w, dw, damp_w):
+    w = w + dw if damp_w > 1e-5 else w
+    return w
+
+
+@gtscript.function
+def ke_from_bwind(ke, ub, vb):
+    return 0.5 * (ke + ub * vb)
+
+
+@gtscript.function
+def adjust_w_and_qcon(w, delp, dw, damp_w, q_con):
+    w = w / delp
+    w = add_dw(w, dw, damp_w)
+    q_con = q_con / delp
+
+    return w, q_con
+
+
 @gtstencil()
-def not_inlineq_pressure(gx: sd, gy: sd, rarea: sd, fx: sd, fy: sd, pt: sd, delp: sd):
+def ke_horizontal_vorticity_w_qcon_adjust(
+    ke: FloatField,
+    u: FloatField,
+    v: FloatField,
+    ub: FloatField,
+    vb: FloatField,
+    ut: FloatField,
+    vt: FloatField,
+    dx: FloatField,
+    dy: FloatField,
+    rarea: FloatField,
+    vorticity: FloatField,
+    w: FloatField,
+    delp: FloatField,
+    dw: FloatField,
+    q_con: FloatField,
+    dt: float,
+    nested: bool,
+    damp_w: float,
+):
+    from __externals__ import (
+        i_end,
+        i_start,
+        j_end,
+        j_start,
+        local_ie,
+        local_is,
+        local_je,
+        local_js,
+    )
+
+    with computation(PARALLEL), interval(...):
+        ke = ke_from_bwind(ke, ub, vb)
+
+        with horizontal(region[i_start, j_start]):
+            if not nested:
+                ke = corners.corner_ke(ke, u, v, ut, vt, dt, 0, 0, -1, 1)
+        with horizontal(region[i_end + 1, j_start]):
+            if not nested:
+                ke = corners.corner_ke(ke, u, v, ut, vt, dt, -1, 0, 0, -1)
+        with horizontal(region[i_end + 1, j_end + 1]):
+            if not nested:
+                ke = corners.corner_ke(ke, u, v, ut, vt, dt, -1, -1, 0, 1)
+        with horizontal(region[i_start, j_end + 1]):
+            if not nested:
+                ke = corners.corner_ke(ke, u, v, ut, vt, dt, 0, -1, -1, -1)
+
+        vt = u * dx
+        ut = v * dy
+        vorticity[0, 0, 0] = rarea * (vt - vt[0, 1, 0] - ut + ut[1, 0, 0])
+
+        with horizontal(region[local_is : local_ie + 1, local_js : local_je + 1]):
+            w, q_con = adjust_w_and_qcon(w, delp, dw, damp_w, q_con)
+
+
+@gtstencil()
+def not_inlineq_pressure(
```

Comment:
can this on could potentially be combined with vbke, but the regions would be awkward, and it wasn't in the list. curiious what you think? I am fine keeping it separate

Reply:
I think it can be combined without too many issues with vbke since the origin is the same and the domain is almost the same.  I created a stencil `not_inlineq_pressure_and_vbke` that combines the two together.